esphome/components/tuya/select/__init__.py

```diff
@@ -0,0 +1,51 @@
+from esphome.components import select
+import esphome.config_validation as cv
+import esphome.codegen as cg
+from esphome.const import CONF_OPTIONS, CONF_OPTIMISTIC, CONF_ID, CONF_ENUM_DATAPOINT
+from .. import tuya_ns, CONF_TUYA_ID, Tuya
+
+DEPENDENCIES = ["tuya"]
+CODEOWNERS = ["@bearpawmaxim"]
+
+TuyaSelect = tuya_ns.class_("TuyaSelect", select.Select, cg.Component)
+
+
+def ensure_option_map():
+    def validator(value):
+        cv.check_not_templatable(value)
+        option = cv.All(cv.int_range(0, 2**8 - 1))
+        mapping = cv.All(cv.string_strict)
+        options_map_schema = cv.Schema({option: mapping})
+        value = options_map_schema(value)
+
+        all_values = list(value.keys())
+        unique_values = set(value.keys())
+        if len(all_values) != len(unique_values):
+            raise cv.Invalid("Mapping values must be unique.")
+
+        return value
+
+    return validator
```

Comment:
Don't really need the validator generator wrapper here
```suggestion
def ensure_option_map(value):
    cv.check_not_templatable(value)
    option = cv.All(cv.int_range(0, 2**8 - 1))
    mapping = cv.All(cv.string_strict)
    options_map_schema = cv.Schema({option: mapping})
    value = options_map_schema(value)

    all_values = list(value.keys())
    unique_values = set(value.keys())
    if len(all_values) != len(unique_values):
        raise cv.Invalid("Mapping values must be unique.")

    return value
```

Reply:
Agreed, removed.